hera_mc/cm_health.py

```diff
@@ -0,0 +1,133 @@
+            connection = [conn.upstream_part, conn.up_part_rev, conn.upstream_output_port,
+                          conn.downstream_part, conn.down_part_rev, conn.downstream_input_port]
+            connection = [x.lower() for x in connection]
+            k = ':'.join(connection)
+            if k in self.conndict.keys():
+                self.conndict[k].append([conn.start_gpstime, conn.stop_gpstime])
+                if k not in self.multiples:
+                    self.multiples.append(k)
+            else:
+                self.conndict[k] = [[conn.start_gpstime, conn.stop_gpstime]]
+
+    def check_for_duplicate_connections(self):
+        if self.conndict is None:
+            self.get_conndict()
+        if len(self.multiples) == 0:
+            print("No duplications found.")
+            return 0
+        duplicates = []
+        for k in self.multiples:
+            for i, t in enumerate(self.conndict[k]):
+                for j in range(i):
+                    start_i = cm_utils.get_astropytime(self.conndict[k][i][0])
+                    stop_i = cm_utils.get_stopdate(self.conndict[k][i][1])
+                    start_j = cm_utils.get_astropytime(self.conndict[k][j][0])
+                    stop_j = cm_utils.get_stopdate(self.conndict[k][j][1])
+                    print("CM_H:  ", k)
+                    if start_j <= start_i:
+                        if stop_j > start_i:
+                            duplicates.append([k, i, j])
+                    else:
+                        if start_j <= stop_i:
+                            duplicates.append([k, i, j])
+        if len(duplicates):
+            print('{} duplications found.'.format(len(duplicates)))
+            for d in duplicates:
+                k = d[0]
+                start_i = cm_utils.get_astropytime(self.conndict[k][i][0])
+                stop_i = cm_utils.get_astropytime(self.conndict[k][i][1])
+                start_j = cm_utils.get_astropytime(self.conndict[k][j][0])
+                stop_j = cm_utils.get_astropytime(self.conndict[k][j][1])
+                print('\t{} <1>{}-{}  <2>{}-{}'.format(k, start_i, stop_i, start_j, stop_j))
+        else:
+            print('No duplications found.')
+        print('Out of {} connections checked.'.format(self.num_connections))
+        return len(duplicates)
+
+    def check_for_existing_connection(self, connection, at_date='now'):
+        at_date = cm_utils.get_astropytime(at_date)
+        connection = [x.lower() for x in connection]
+        k = ':'.join(connection)
+        if self.conndict is None:
+            self.get_conndict()
+        if k not in self.conndict.keys():
+            return False
+        for t in self.conndict[k]:
+            start_t = cm_utils.get_astropytime(t[0])
+            stop_t = cm_utils.get_astropytime(t[1])
+            if cm_utils.is_active(at_date, start_t, stop_t):
+                print('Duplicate found for connection {} at {}:  {} - {}'.format(k, at_date, start_t, stop_t))
```

Comment:
If this function returns True, it doesn't necessarily indicate a *duplicate* connection, right? I.e., should this printed message be changed?